src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_inpaint_legacy.py

```diff
@@ -44,17 +45,26 @@ def preprocess_image(image):
     return 2.0 * image - 1.0
 
 
+# add additional output for mask tensor corresponding to input image (in addition to latent mask)
 def preprocess_mask(mask, scale_factor=8):
     mask = mask.convert("L")
     w, h = mask.size
     w, h = map(lambda x: x - x % 32, (w, h))  # resize to integer multiple of 32
+
+    input_mask = np.array(mask).astype(np.float32) / 255.0
```

Comment:
This looks a bit backwards breaking - what are we doing here exactly?

Reply:
Could you provide a code snippet explaining why this is needed maybe? :-)

Reply:
The input_mask keeps track of the drawn mask, so that after decoding the inpainted image, we can replace the non-masked parts of the output image with the original image, and only replace the masked portions of the output image with the generated inpainted image. 
It's not absolutely necessary, it just addresses an issue I've seen especially noticeable with anime style sd checkpoints, where changing a small part of the latents (corresponding to the drawn mask) causes distortion of other parts of the decoded image, even far from the drawn image, due to the global nature of the VAE decoder.

Reply:
I pasted example differences below showing the effect of preserving the original image (first screenshot, colors match) vs using the full inpainted image (colors slightly grayed). On second thought the distortion away from masked area may largely be due to the VAE reconstruction loss. The webui code for the examples is [here](https://github.com/1lint/sdpipe_webui)

<img width="1920" alt="preserve_nonmasked_original" src="https://user-images.githubusercontent.com/105617163/214112033-e90a5a03-9618-4a95-9e15-dec26f7acd92.png">
<img width="1920" alt="entire_inpainted_image" src="https://user-images.githubusercontent.com/105617163/214112017-4ea0d4f3-5888-4b46-a3ab-5b247cdf11a1.png">